server/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -739,6 +740,6 @@ public void cancelTaskAndDescendants(CancellableTask task, String reason, boolea
     }
 
     public List<String> getTaskHeaders() {
-        return taskHeaders;
+        return List.of(taskHeaders);
```

Comment:
Is it possible to replace `ArrayList` above with an `List.of` instead? This way we would not keep mutable array and will not create list every time somebody calls a getter.

Reply:
I figured this is on the hot path and it's nice to really get rid of any and all allocations for the iteration over the list here. This method is used by (a) test only anyway :)